### ISSUE TITLE
Moving repo to a shared repo?

### ISSUE THREAD

**Issue by feens** [gazoodle/geckolib#47]
It looks like work on this repo isn't really happening anymore.  I'm curious if there's a way we can move development to a shared repo that isn't owned by a specific person.

**Comment by feens** [comment]
@gazoodle I'm not sure the exact process to be honest...I've reached out to a few people who have forked off of this repo and started making changes as well, seeing if we can find a way to combine efforts and make sure that the future of the project isn't reliant on any single person. 

Also, definitely not a slight against you or anything...I think it's the downside to making something that other people end up interested in, and it's just a side project. I know my interests and hobbies are always shifting around in terms of the time I can give them.

**Comment by feens** [comment]
A quick bit of reading tells me one option is to create an organization/repo.  It _may_ also be ideal to move eventually the home assistant repo into an official integration but I'm not sure what's required for that, and it doesn't answer the question as to what to do with the library repo. 

If others come back with interest I can look into setting up the org/repo.  Maybe we could work together on getting some more comprehensive documentation together for those looking to develop as well.

**Comment by gazoodle** [comment]
> @gazoodle I'm not sure the exact process to be honest...I've reached out to a few people who have forked off of this repo and started making changes as well, seeing if we can find a way to combine efforts and make sure that the future of the project isn't reliant on any single person.

I totally agree. 

> 
> Also, definitely not a slight against you or anything...I think it's the downside to making something that other people end up interested in, and it's just a side project. I know my interests and hobbies are always shifting around in terms of the time I can give them.

I've got no ego when it comes to collaboration on code, and like you said it's best if this isn't reliant on a single person/point of failure. I'm happy to contribute where I can and obviously the code is written in my style according to how my brain was thinking at the time, but I've no real sense of ownership on it, merely that the stuff works and does what I want it do. If it's also useful to others (I long ago learnt I'm not unique in my requirements) then I'm happy to share. Home Assistant is such a great project and I get so much value from it, I'm more than happy to give back in any way I can, Nabu Casa subscriptions and of course contributing libraries and integrations for the gear that I'm using.

**Comment by gazoodle** [comment]
> A quick bit of reading tells me one option is to create an organization/repo. It _may_ also be ideal to move eventually the home assistant repo into an official integration but I'm not sure what's required for that, and it doesn't answer the question as to what to do with the library repo.

I looked into doing this a while ago, but the requirements to get to a reasonable standard were beyond me at the time. I reached out to the author of the Bilbao integration that is part of the core HA project and took on board the comments.

HA is very strict when it comes to protocol libraries not being part of the HA code base, hence why it's split into geckolib (the bit that does the comms) and the integration which simply consumes the library and builds the facade for HA to interact with the library.

> 
> If others come back with interest I can look into setting up the org/repo. Maybe we could work together on getting some more comprehensive documentation together for those looking to develop as well.

Happy to help document what I've found out with endless hours of Wireshark sniffing what goes on when buttons are pressed on the in.touch2 app. Now I don't need to go down to that level because the library has a simulator module as part of geckolib which I've used to ensure that I can replicate behaviour in gecko modules that I don't have access to.

As I alluded to earlier, I got sucked into another mammoth project for the last 18 months and I've not been able to spend any time on this, much to my chagrin. This ought to change in the next couple of months.

I've noticed that there are some pull requests which I'm happy to enact, I just need to refresh how to get the library release to PyPI after that and then remember how to release a new integration :-)

**Comment by feens** [comment]
@gazoodle so after some more digging, an organization is the way to go.  I looked at setting it up, in which case I think it ends up making me the creator of the org, so it may be better for you to tackle that.  The advantage of having an org is there can be multiple owners/maintainers/etc.  My hope is that it takes the pressure off of you and makes it easier to collaborate with anyone else interested in ensuring the project moves along.  

A good example of this would be [ratgdo](https://github.com/ratgdo), a project for Chamberlain/Liftmaster controller devices and software.  The creator has a [page in GH](https://github.com/PaulWieland/ratgdo), but the associate repositories all live under the org.

**Comment by gazoodle** [comment]
@feens I've started the process, created an org and I've been playing around with migrating the repos. I'm busy in the next couple of weeks, but hope to get them migrated after that.

**Comment by feens** [comment]
@gazoodle awesome, appreciate you doing that.

**Comment by M0LTE** [comment]
I don't think this happened in the end?

**Comment by BenJewell** [comment]
> HA is very strict when it comes to protocol libraries not being part of the HA code base, hence why it's split into geckolib (the bit that does the comms) and the integration which simply consumes the library and builds the facade for HA to interact with the library.

Yeah I noticed other integrations doing that, but it makes it hard to modify them I think. In one case on another project, I forked the library into a different name in Pypi. I modified the requirements file in the integration, but Home Assistant was somehow caching it and would not take the changes. I could never figure that out.

Gave up after a couple hours and went with a hacky solution. I ended up just writing an automation that runs every time Home Assistant restarts, which pulls down my library files.
